arguments/MultiQueue.argdown

```diff
@@ -0,0 +1,55 @@
+## Multiple GPU Queues
+<!--
+Expose more than a single GPU queue:
+  - investigation: https://github.com/gpuweb/gpuweb/issues/1065
+  - proposal-1: https://github.com/gpuweb/gpuweb/issues/1066
+  - proposal-2: https://github.com/gpuweb/gpuweb/issues/1073
+-->
+
+[Expose API]: We should expose multiple queues with different capabilities.
+  + <API Support>: Common across all three APIs
+               (Metal starting with version 2, where `MTLEvent` got introduced).
+  + <Performance>: Async compute and transfers are established best practices on
+                   the last-gen console hardware (PS4 and XB1).
+                   See [AMD presentation](https://gpuopen.com/wp-content/uploads/2017/03/GDC2017-Asynchronous-Compute-Deep-Dive.pdf).
+  + <Viable>: Investigation showed that we can safely track the ownership of resources
+              by the queues, and insert the necessary pipeline barriers, making this safe.
+  +> [Implicit Transitions]
+  +> [Required Transitions]
+
+[Fully Internal]: WebGPU should internally use multiple queues but not necessarily
+                  expose them to the user.
+  + <Simple>: Simplest to use.
+  - <Ineffective>: WebGPU doesn't know resource dependencies ahead of time, so finding
+                   the efficient scheduling route is hard. It's likely to synchronize
+                   more than necessary.
+  + <Metal>: this is what Metal-1 does in the drivers/run-time.
+  -> [Expose API]
+
+[Implicit Transitions]: Ownership transition can happen automatically, based on resources
+                        that are used by submissions.
+  + <No Fence>: Fences are technically not needed in the API and can be removed?
```

Comment:
I'm not sure this is a +, maybe more of a note. (we'll still need an API to watch for completion)

Reply:
But mapping gives you that completion if you need it. We haven't gotten any other use cases for fences, have we?

Reply:
Sure, mapping gives you completion but it's a bit roundabout. Apart from multiqueue we don't have another usecase for fences.